Source/WebCore/platform/graphics/SourceBufferPrivate.cpp

```diff
@@ -1083,29 +1166,33 @@ void SourceBufferPrivate::didReceiveSample(Ref<MediaSample>&& originalSample)
     updateHighestPresentationTimestamp();
 }
 
-void SourceBufferPrivate::append(Ref<SharedBuffer>&& buffer)
+void SourceBufferPrivate::processMediaSamples()
```

Comment:
Passing an operation as parameter would remove the ASSERT and would probably be more readable.